src/Domain/Value/Resolver/RelationCollection.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of Storyblok-Api.
+ *
+ * (c) SensioLabs Deutschland <info@sensiolabs.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Storyblok\Api\Domain\Value\Resolver;
+
+/**
+ * @author Silas Joisten <silasjoisten@proton.me>
+ *
+ * @implements \IteratorAggregate<int, Relation>
+ */
+final class RelationCollection implements \Countable, \IteratorAggregate
+{
+    /**
+     * @var list<Relation>
+     */
+    private array $items = [];
+
+    /**
+     * @param list<Relation> $items
+     */
+    public function __construct(
+        array $items = [],
+    ) {
+        foreach ($items as $item) {
+            $this->add($item);
+        }
+    }
```

Comment:
Do we really want to allow an empty RelationCollection?

Reply:
Yes you can.

Reply:
Yes I know, that's written down in the code, let me rephrase. IF a collection can be empty, then we don't need to allow `null` and don't need to use it as a default value in `StoriesRequest`, just use:

```diff
- public ?RelationCollection $withRelations = null,
+ public RelationCollection $withRelations = New RelationCollection(),
```

this would also simplify the check in `StoriesRequest::toArray()` then

Reply:
Indeed i agree. I would do that in a follow up PR i guess?